torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -1075,4 +1095,25 @@ at::Scalar PythonArgs::scalar_slow(int i) {
   return at::Scalar(THPUtils_unpackDouble(args[i]));
 }
 
+at::Scalar PythonArgs::scalar_slow(PyObject* arg) {
```

Comment:
What's going on here? How come the implementation above isn't usable?

Reply:
The difference between these two is we are passing `PyObject* arg` instead of `args` index. In [PythonArgs::scalarlist(int i)](https://github.com/pytorch/pytorch/pull/48222/files#diff-fd606fc3774dfe35a405733d24991eb8602da209887beeb77dc0f91e6f669695R292) we are checking if `args[i]` is a tuple or a list, wrap it into the python object and pass it down. The original [PythonArgs::scalar_slow(int i)](https://github.com/pytorch/pytorch/pull/48222/files#diff-bb8fc823d2ddc10e38cabfc113c4aa0682b6082089079209abab27b7715d1e3eR1071) isn't doing this, which results in different behavior inside of [THPUtils_unpackDouble](https://github.com/pytorch/pytorch/blob/88ebf6f894a61039d2ac0077438b0ad3637c7a71/torch/csrc/utils/python_numbers.h#L113). In case if we pass `args[i]` (original scalar_slow(int i)), we get `TypeError: must be real number, not list`, but in case of python object, we return `PyFloat_AS_DOUBLE`

Reply:
Then still the function above should be calling into this one, replaing the logic on lines 1080-1095?

Reply:
done